src/displayapp/screens/settings/SettingPineTimeStyle.cpp

```diff
@@ -0,0 +1,332 @@
+      lv_obj_set_style_local_text_color(timeDD1, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+      lv_obj_set_style_local_text_color(timeDD2, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+      lv_obj_set_style_local_text_color(timeAMPM, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueTime]);
+    }
+  }
+  if((object == btnNextBar) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBar < 16 ) {
+      valueBar += 1;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    } else {
+      valueBar = 0;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    }
+  }
+  if((object == btnPrevBar) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBar > 0 ) {
+      valueBar -= 1;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    } else {
+      valueBar = 16;
+      settingsController.SetPTSColorBar(valueBar);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBar]);
+    }
+  }
+  if((object == btnNextBG) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBG < 16 ) {
+      valueBG += 1;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    } else {
+      valueBG = 0;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    }
+  }
+  if((object == btnPrevBG) && (event == LV_EVENT_PRESSED)) {
+    if ( valueBG > 0 ) {
+      valueBG -= 1;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    } else {
+      valueBG = 16;
+      settingsController.SetPTSColorBG(valueBG);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[valueBG]);
+    }
+  }
+  if((object == btnReset) && (event == LV_EVENT_PRESSED)) {
+      settingsController.SetPTSColorTime(11);
+      lv_obj_set_style_local_text_color(timeDD1, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[11]);
+      lv_obj_set_style_local_text_color(timeDD2, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[11]);
+      lv_obj_set_style_local_text_color(timeAMPM, LV_LABEL_PART_MAIN, LV_STATE_DEFAULT, pts_colors[11]);
+      settingsController.SetPTSColorBar(11);
+      lv_obj_set_style_local_bg_color(sidebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[11]);
+      settingsController.SetPTSColorBG(3);
+      lv_obj_set_style_local_bg_color(timebar, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, pts_colors[3]);
```

Comment:
There should probably be an enum for pts_colors[].

Reply:
I'm afraid I don't know what an enum is or what advantage it offers, but the main reason it's an array is because i only store the array index in settings rather than the color

Reply:
With an enum, this could be `pts_colors[colorBlack]` and `SetPTSColorBG(colorBlack)` for example, which is clearer than 3 or 11,  but there may be some even better way to do it.